node/commons/compatibility-layer.ts

```diff
@@ -279,6 +287,16 @@ const fillSearchItemWithSimulation = (searchItem: SearchItem, orderFormItems: Or
         PaymentSystemGroupName: '',
       }]
     })
+
+    const sellerDefault = getSellerDefault(searchItem.sellers)
+    if (sellerDefault && searchItem.sellers.find(seller => seller.sellerDefault)?.sellerId !== sellerDefault) {
+      searchItem.sellers = searchItem.sellers.map(seller => {
+        return {
+          ...seller,
+          sellerDefault: seller.sellerId === sellerDefault ? true : false
+        }
+      })
+    }
   }
```

Comment:
what do think about creating a new function `fixSellerDefaultAccordingToStock`? I think this function will help someone that is reading the code but doesn't need to understand how we do that.